packages/ui/src/molecules/SearchInput/SearchInput.tsx

```diff
@@ -61,25 +66,35 @@ const SearchInput = forwardRef<HTMLFormElement, SearchInputProps>(
     },
     ref
   ) {
-    const valueRef = useRef<HTMLInputElement>(null)
+    const inputRef = useRef<HTMLInputElement>(null)
+    const formRef = useRef<HTMLFormElement>(null)
 
     const handleSubmit = (event: FormEvent) => {
       event.preventDefault()
 
-      if (valueRef.current?.value !== '') {
-        onSubmit(valueRef.current!.value)
+      if (inputRef.current?.value !== '') {
+        onSubmit(inputRef.current!.value)
       }
     }
 
+    useImperativeHandle(ref, () => ({
+      get inputRef() {
+        return inputRef.current
+      },
+      get formRef() {
+        return formRef.current
+      },
```

Comment:
Would the following also work? Feels simpler.

```suggestion
      inputRef: inputRef.current,
      formRef: formRef.current,
```